### ISSUE TITLE
Base64 - rewrite to follow MISRA C:2012

### ISSUE THREAD

**Issue by Igor-Misic** [IMProject/IMBootloader#13]
Since we are pursuing to satisfy **MISRA C:2012** we have a chance to be the first project that has base64 in that format. 
Before we jump and do it with the existing Base64 code there are a couple of things to consider. There are many different Base64 implementations and each has a different execution speed. At the moment we are using one that is not the fastest but was the easiest to integrate into our project. Here is a good point explaining which implementation is the fastest:
https://stackoverflow.com/questions/342409/how-do-i-base64-encode-decode-in-c

In short, we should consider using **jounimalinen** for encoding and **polfosol** for decoding. 
code for both can be found here:
https://github.com/aklomp/base64

Base64 is now part of IMUtility repo:
https://github.com/IMProject/IMUtility

**Comment by Igor-Misic** [comment]
Thanks @jciberlin ! :car: power! :smiley:

**Comment by Igor-Misic** [comment]
Thanks @jciberlin ! :car: power!   :smiley:
